dsl/src/test/scala/org/http4s/dsl/PathSpec.scala

```diff
@@ -12,31 +12,27 @@ package dsl
 
 import cats.effect.IO
 import org.http4s.Uri.uri
+import org.http4s.Uri.Path
+import org.http4s.Uri.Path.{Root, Segment}
 import org.http4s.dsl.io._
-import org.scalacheck.Arbitrary
 import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
 
 class PathSpec extends Http4sSpec {
-  implicit val arbitraryPath: Arbitrary[Path] =
-    Arbitrary {
-      arbitrary[List[String]].map(Path(_))
-    }
+  implicit val arbitraryPath: Gen[Path] =
+    arbitrary[List[String]]
+      .map(_.foldLeft(Path.Root)((acc, e) => acc.copy(segments = acc.segments :+ Segment(e))))
```

Comment:
`acc.addSegment(e)` or `acc / e`

that should eliminate both uses of copy

Reply:
done